From 3cb7c8fb18e76cb9d02935871b531b4cfd22bfb8 Mon Sep 17 00:00:00 2001
From: Paolo Sabatino <paolo.sabatino@gmail.com>
Date: Fri, 30 Jan 2026 22:58:25 +0100
Subject: [PATCH] rockchip: declare back rockchip gpio non-sleeping

 * linux kernel commit 20cf2aed89ac6d78a0122e31c875228e15247194
   declared rockchip GPIO driver as sleeping because some
   helpers use mutexes in shared GPIO context. We declare
   gpio driver non-sleeping as it was before because 32 bit
   rockchip platforms should not use shared gpios so far.
   This fixes at least huge number of warnings about
   gpiod_get_value() call when gpio-ir-recv remote controller
   driver receives input
---
 drivers/gpio/gpio-rockchip.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpio/gpio-rockchip.c b/drivers/gpio/gpio-rockchip.c
index bae2061f15fc..47174eb3ba76 100644
--- a/drivers/gpio/gpio-rockchip.c
+++ b/drivers/gpio/gpio-rockchip.c
@@ -593,7 +593,6 @@ static int rockchip_gpiolib_register(struct rockchip_pin_bank *bank)
 	gc->ngpio = bank->nr_pins;
 	gc->label = bank->name;
 	gc->parent = bank->dev;
-	gc->can_sleep = true;
 
 	ret = gpiochip_add_data(gc, bank);
 	if (ret) {
-- 
2.43.0

